test/opt/opextinst_forward_ref_fixup_pass_test.cpp

```diff
@@ -0,0 +1,365 @@
+                  %27 = OpExtInst %void %1 DebugLocalVariable %8 %21 %16 %uint_4 %uint_3 %25 %uint_288 %uint_1
+                  %29 = OpExtInst %void %1 DebugTypeFunction %uint_3 %void
+                  %30 = OpExtInst %void %1 DebugFunction %9 %29 %16 %uint_9 %uint_1 %17 %7 %uint_3 %uint_9
+                  %32 = OpExtInst %void %1 DebugLexicalBlock %16 %uint_9 %uint_13 %30
+                  %34 = OpExtInst %void %1 DebugLocalVariable %10 %21 %16 %uint_10 %uint_5 %32 %uint_4
+                  %36 = OpExtInst %void %1 DebugEntryPoint %30 %17 %11 %12
+                  %37 = OpExtInst %void %1 DebugInlinedAt %uint_11 %32
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugExpression
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugSource
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugCompilationUnit
+; CHECK:       {{.*}} = OpExtInstWithForwardRefsKHR %void {{.*}} DebugTypeComposite
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugTypeFunction
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugFunction
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugLocalVariable
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugTypeFunction
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugFunction
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugLexicalBlock
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugLocalVariable
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugEntryPoint
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugInlinedAt
+                %main = OpFunction %void None %40
+                  %43 = OpLabel
+                  %44 = OpVariable %_ptr_Function_A Function
+                  %45 = OpExtInst %void %1 DebugFunctionDefinition %30 %main
+                  %51 = OpExtInst %void %1 DebugScope %32
+                  %46 = OpExtInst %void %1 DebugLine %16 %uint_10 %uint_10 %uint_3 %uint_5
+                  %47 = OpExtInst %void %1 DebugDeclare %34 %44 %15
+                  %52 = OpExtInst %void %1 DebugScope %25 %37
+                  %48 = OpExtInst %void %1 DebugLine %16 %uint_4 %uint_5 %uint_3 %uint_3
+                  %49 = OpExtInst %void %1 DebugDeclare %27 %44 %15
+                  %53 = OpExtInst %void %1 DebugNoScope
+                  %50 = OpExtInst %void %1 DebugLine %16 %uint_12 %uint_12 %uint_1 %uint_1
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugFunctionDefinition
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugScope
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugLine
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugDeclare
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugScope
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugLine
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugDeclare
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugNoScope
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugLine
+                        OpReturn
+                        OpFunctionEnd
+  )";
+  const auto result =
+      SinglePassRunAndMatch<OpExtInstWithForwardReferenceFixupPass>(
```

Comment:
Again you could do something like this:

https://github.com/KhronosGroup/SPIRV-Tools/blob/65d30c31508e47d34f7698f511aea4800e3d44fe/test/opt/redundancy_elimination_test.cpp#L359